### ISSUE TITLE
Allow map + dispatch

### ISSUE THREAD

**Issue by whiskeysierra** [zalando/riptide#71]
```java
private URI create(URI url, String body) {
    return unit.execute(POST, url, body).dispatch(series(),
            on(SUCCESSFUL).map(normalize(url)).dispatch(sameHeader("Location"),
                    on("Content-Location").capture(),
                    any(String.class).capture(location().andThen(location ->
                            unit.execute(GET, location).dispatch(series(),
                                    on(SUCCESSFUL).capture(String.class),
                                    anySeries().call(this::fail))
                                    .to(String.class)))),
            anySeries().call(this::fail))
            .to(URI.class);
}
```

**Comment by whiskeysierra** [comment]
Would require to re-introduce `map`...

**Comment by whiskeysierra** [comment]
Or alternatively:

```java
private URI create(URI url, String body) {
    return unit.execute(POST, url, body).dispatch(series(),
            on(SUCCESSFUL).dispatch(normalize(url), isCurrentRepresentation(),
                    on(true).capture(String.class),
                    on(false).capture(location().andThen(location ->
                            unit.execute(GET, location).dispatch(series(),
                                    on(SUCCESSFUL).capture(String.class),
                                    anySeries().call(this::fail))
                                    .to(String.class)))),
            anySeries().call(this::fail))
            .to(URI.class);
}
```

**Comment by whiskeysierra** [comment]
Would be more consistent with `call` and `capture`.

**Comment by whiskeysierra** [comment]
Fixed with #75
